llvm/tools/sycl-post-link/sycl-post-link.cpp

```diff
@@ -313,8 +313,12 @@ std::vector<StringRef> getKernelNamesUsingAssert(const Module &M) {
 }
 
 bool isModuleUsingAsan(const Module &M) {
-  auto *AsanInitFunction = M.getFunction("__asan_init");
-  return AsanInitFunction;
+  for (auto &F : M) {
+    if (F.getFunction().getName().starts_with("__asan_")) {
+      return true;
+    }
+  }
+  return false;
```

Comment:
```suggestion
  return llvm::any_of(M.functions(), [](const Function *F) {
    return F.getName().startswith("__asan_");
  });
```

If we know that `__asan_` functions are only inserted as declarations, then I would suggest to also add a check if a function is a declaration, because it would speed up the search by skipping all user-provided definitions

Reply:
In post link stage, I think the kernel has already linked to libdevice, so they're definition now.